inst/maintenance/unit/test-mread.R

```diff
@@ -127,10 +127,26 @@ test_that("Error when code is passed as project", {
 })
 
 test_that("Model name with spaces is error", {
-    expect_error(mcode("ab cd", ""))
+  expect_error(mcode("ab cd", ""))
 })
 
 test_that("Error with duplicate blocks", {
   expect_error(mcode("a", "$MAIN \n $MAIN",compile = FALSE))
   expect_error(mcode("a", "$SET \n $SET",compile = FALSE))
 })
+
+test_that("Recover data when compile fails", {
+  code <- '[main] double a = 2\n[param] b = 5\n'
+  expect_warning(
+    mod <- mcode("fail", code, recover = TRUE), 
+    regexp = "returning object for debugging purposes only"
+  )
+  expect_is(mod, "list")
+  expect_named(mod)
+  expect_true("mod" %in% names(mod))
+  expect_true("build" %in% names(mod))
+  expect_true("out" %in% names(mod))
+  expect_true("spec" %in% names(mod))
+  recov <- mrgsolve:::build_format_recover(data)
```

Comment:
Should the `data` object here be the output of trying to compile this model? Right now this is returning `utils::data()` as character (so the test passes, but maybe this isn't doing what you're aiming for).

Reply:
Doh! Will fix it.